File: app/src/main/java/com/niceplaces/niceplaces/utils/TextUtils.java
package com.niceplaces.niceplaces.utils;

import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.text.SpannableString;
import android.text.SpannableStringBuilder;
import android.text.Spanned;
import android.text.TextPaint;
import android.text.style.ImageSpan;
import android.widget.TextView;

import java.util.concurrent.atomic.AtomicBoolean;

public class TextUtils {

    public static void justify(final TextView textView) {

        final AtomicBoolean isJustify = new AtomicBoolean(false);

        final String textString = textView.getText().toString();

        final TextPaint textPaint = textView.getPaint();

        final SpannableStringBuilder builder = new SpannableStringBuilder();

        textView.post(new Runnable() {
            @Override
            public void run() {

                try {

                    if (!isJustify.get()) {

                        final int lineCount = textView.getLineCount();
                        final int textViewWidth = textView.getWidth();

                        for (int i = 0; i < lineCount; i++) {

                            int lineStart = textView.getLayout().getLineStart(i);
                            int lineEnd = textView.getLayout().getLineEnd(i);

                            String lineString = textString.substring(lineStart, lineEnd);

                            if (i == lineCount - 1) {
                                builder.append(new SpannableString(lineString));
                                break;
                            }

                            String trimSpaceText = lineString.trim();
                            String removeSpaceText = lineString.replaceAll(" ", "");

                            float removeSpaceWidth = textPaint.measureText(removeSpaceText);
                            float spaceCount = trimSpaceText.length() - removeSpaceText.length();

                            float eachSpaceWidth = (textViewWidth - removeSpaceWidth) / spaceCount;

                            SpannableString spannableString = new SpannableString(lineString);
                            for (int j = 0; j < trimSpaceText.length(); j++) {
                                char c = trimSpaceText.charAt(j);
                                if (c == ' ') {
                                    Drawable drawable = new ColorDrawable(0x00ffffff);
                                    drawable.setBounds(0, 0, (int) eachSpaceWidth, 0);
                                    ImageSpan span = new ImageSpan(drawable);
                                    spannableString.setSpan(span, j, j + 1, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
                                }
                            }

                            builder.append(spannableString);
                        }

                        textView.setText(builder);
                        isJustify.set(true);
                    }

                } catch (StringIndexOutOfBoundsException e) {
                    e.printStackTrace();
                }
            }
        });
    }


}

File: app/src/main/java/com/niceplaces/niceplaces/utils/TextUtils.kt
package com.niceplaces.niceplaces.utils

import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable
import android.text.SpannableString
import android.text.SpannableStringBuilder
import android.text.Spanned
import android.text.style.ImageSpan
import android.widget.TextView
import java.util.concurrent.atomic.AtomicBoolean

object TextUtils {
    fun justify(textView: TextView) {
        val isJustify = AtomicBoolean(false)
        val textString = textView.text.toString()
        val textPaint = textView.paint
        val builder = SpannableStringBuilder()
        textView.post {
            try {
                if (!isJustify.get()) {
                    val lineCount = textView.lineCount
                    val textViewWidth = textView.width
                    for (i in 0 until lineCount) {
                        val lineStart = textView.layout.getLineStart(i)
                        val lineEnd = textView.layout.getLineEnd(i)
                        val lineString = textString.substring(lineStart, lineEnd)
                        if (i == lineCount - 1) {
                            builder.append(SpannableString(lineString))
                            break
                        }
                        val trimSpaceText = lineString.trim { it <= ' ' }
                        val removeSpaceText = lineString.replace(" ".toRegex(), "")
                        val removeSpaceWidth = textPaint.measureText(removeSpaceText)
                        val spaceCount = trimSpaceText.length - removeSpaceText.length.toFloat()
                        val eachSpaceWidth = (textViewWidth - removeSpaceWidth) / spaceCount
                        val spannableString = SpannableString(lineString)
                        for (j in 0 until trimSpaceText.length) {
                            val c = trimSpaceText[j]
                            if (c == ' ') {
                                val drawable: Drawable = ColorDrawable(0x00ffffff)
                                drawable.setBounds(0, 0, eachSpaceWidth.toInt(), 0)
                                val span = ImageSpan(drawable)
                                spannableString.setSpan(span, j, j + 1, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
                            }
                        }
                        builder.append(spannableString)
                    }
                    textView.text = builder
                    isJustify.set(true)
                }
            } catch (e: StringIndexOutOfBoundsException) {
                e.printStackTrace()
            }
        }
    }
}